### ISSUE TITLE
Android toggle Tile 

### ISSUE THREAD

**Issue by x8BitRain** [Jigsaw-Code/outline-client#32]
As an enhancement, it would be nice and convenient to have a tile button you can put into your quick settings drawer on Android. ShadowSocks has this feature already.

**Comment by trevj** [comment]
Hi @x8BitRain Thanks for the suggestion. We'll take it into consideration.

**Comment by laosb** [comment]
iOS Widget too, please.

**Comment by WarenGonzaga** [comment]
Is this still active?

**Comment by Sominemo** [comment]
No updates?

**Comment by Goodsmileduck** [comment]
Why it was closed?

**Comment by Kiura** [comment]
Any updates on this?

**Comment by stlr00** [comment]
Still need this

**Comment by Simulping** [comment]
definitely needed, really annoying how I can't quickly enable it as opposed to using Wireguard

**Comment by trolik** [comment]
Still need this

**Comment by osipxd** [comment]
Are you accepting pull requests?

**Comment by Garik-** [comment]
Still need this

**Comment by the-webb-identity** [comment]
@ohnorobo any way to bump this up?

**Comment by ohnorobo** [comment]
@the-webb-identity I don't think this is something we'll be able to prioritize soon. But since it involves a small separate system I do think it would be promising to have an external contributor work on without having to disturb the VPN internals.

The parts I can see would be:
- figuring out the android tile system (this will be much easier nowadays working with AI)
- creating the toggle UI (this could re-use some of the existing UI for users with a single key)
- wiring up to the existing startVpn / stopVpn
- various testing details around handling users with edge cases like multiple servers.

**Comment by TurboTheTurtle** [comment]
@ohnorobo I opened a PR for this in #2767. It adds an Android Quick Settings tile that toggles the last successful tunnel and keeps the expanded tile state in sync with the VPN status.
